### ISSUE TITLE
When run train.py. IndexError:index 27 is out of bounds for axis 1 with size 13

### ISSUE THREAD

**Issue by jinbooooom** [qqwweee/keras-yolo3#42]
When I trained my own data,ran train.py,this error occurred.

![question](https://user-images.githubusercontent.com/33092119/39752677-bfe0b34e-52ee-11e8-9816-a14b500e55cc.png)

I think there is no problem with these data, because I have successfully trained in yolo2. I also modified the classes and filters in the configuration file.
Then run python convert.py my_yolo3.cfg yolov3.weights model_data/yolo.h5.This command is no problem.But when I run train.py,this error occurred.
Is it my operation that leads to this problem?Thank you.

**Comment by qqwweee** [comment]
Raw training data is a picture, and bounding boxes according to the size of picture. You define them in annotation file.
The error should not occur.

**Comment by yinggo** [comment]
I also meet the similar problem, can you give more details about how to define the original pic size in annotation file, and how to modify the corresponding code? Thanks! @qqwweee

**Comment by jinbooooom** [comment]
@qqwweee  
Thank you for your reply！But I still haven't solved this problem.
I noticed that when I use ''**python convert.py yolov3.cfg yolov3.weights model_data/yolo.h5**'' command in my own data(classes=3),terminal print out:
**Total params: 61,587,112**
**Trainable params: 61,534,504**
**Non-trainable params: 52,608**
__________________________________________________________________________________________________
**None**
**Saved Keras model to model_data/yolo.h5**
**Read 61587112 of 62001757.0 from Darknet weights.**
**Warning: 414645.0 unused weights**

When I use that command in Pascal VOC Challenge data,print:

**Total params: 62,001,757**
**Trainable params: 61,949,149**
**Non-trainable params: 52,608**
__________________________________________________________________________________________________
**None**
**Saved Keras model to model_data/yolo.h5**
**Read 62001757 of 62001757.0 from Darknet weights.**
So I failed to convert Darknet weights file into yolo.h5 file???
Your reply in this issue(#29):
''**If you have a model that can detect more than 3 classes ["person","bicycle","car"], then you can clip the weights to just detect these 3 classes. That is, output conv layers have 3*(5+num_classes) filters, just use model.get_weights() to get the numpy array, choose some dims and load it on new model.
It is OK because yolo3 uses sigmoid as class probability function instead of softmax on yolo2.**''
Would you tell me how to use **model.get_weights()**?Thank You very much!

**Comment by qqwweee** [comment]
The two weights files both have 62001757 weights. It seems that you used the same file. The 3 classes weights file is a little smaller. 62001757 is coco_classes weights.

**Comment by jinbooooom** [comment]
I see,Thank you very much

**Comment by wizholy** [comment]
do you fix this question?

**Comment by king-cz** [comment]
None
Saved Keras weights to model_data/yolo_weights.h5
Read 61581727 of 62001757.0 from Darknet weights.
Warning: 420030.0 unused weights
I can't solve this problem

**Comment by stephenbhongms** [comment]
I met the similar situation when len(annotation_lines) is NOT multiplying with 100.  The reason is because i = (i+1) % n in  "def bottleneck_generator( ...)" produces  index with out of bound. Just modify this line code as i =(i+1)% n -1 for any length of len(annotation_lines)(i.e., any length of training images). Wish  it helps.
